Firestore/Source/API/FIRFirestoreSettings.mm

```diff
@@ -54,7 +54,10 @@ - (BOOL)isEqual:(id)other {
          self.isSSLEnabled == otherSettings.isSSLEnabled &&
          self.dispatchQueue == otherSettings.dispatchQueue &&
          self.isPersistenceEnabled == otherSettings.isPersistenceEnabled &&
+#pragma clang diagnostic push
```

Comment:
This could be slightly cleaner if you added something like this near the top:

```c++
inline BOOL TimestampsInSnapshotsEnabled(FIRFirestoreSettings *settings) {
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
  return settings.timestampsInSnapshotsEnabled;
#pragma clang diagnostic pop
}
```

... and then used the helper throughout.

This isn't terribly important here, but in general C++ code this kind of suppression usually requires something special for each supported compiler and a helper is much more important in that case.

Optional.

Reply:
Thanks, I did consider a helper, but we also set the property, so I'd need 2 helpers.  And then there are also a couple usages in FIRDocumentSnapshot which couldn't use the helpers (unless I exposed them in a _Private.h header or something)...  So I decided to just live with the suppressions. Hopefully they don't live too long.